ParamspecArgs cannot be used with enumerate

**Bug Report**

With https://github.com/python/mypy/issues/12386 an issue was closed, that ParamSpec cannot be used as tuples and dicts. However, enumerate is seen as invalid, even if it works.

**To Reproduce**

```py
from typing_extensions import ParamSpec
from typing import Callable, TypeVar

P = ParamSpec("P")
R = TypeVar("R")

def foo(func: Callable[P, R]) -> Callable[P, R]:
    def _inner(*args: P.args, **kwargs: P.kwargs) -> R:
        x = kwargs.pop("x")
        y = enumerate(args)
        return func(*args, **kwargs)
    return _inner
```

**Expected Behavior**

It works fine

**Actual Behavior**

```
foo.py:10:13: error: Need type annotation for "y"  [var-annotated]
            y = enumerate(args)
                ^
foo.py:10:23: error: Argument 1 to "enumerate" has incompatible type "P.args"; expected "Iterable[<nothing>]"  [arg-type]
            y = enumerate(args)
                          ^
Found 2 errors in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 0.960
- Python version used: 3.7.2
- Operating system and version: SUSE Linux

--- a/mypy/constraints.py
+++ b/mypy/constraints.py
@@ -571,6 +571,8 @@ class ConstraintBuilderVisitor(TypeVisitor[List[Constraint]]):
             if not actual.values:
                 return infer_constraints(template, actual.upper_bound, self.direction)
             return []
+        elif isinstance(actual, ParamSpecType):
+            return infer_constraints(template, actual.upper_bound, self.direction)
         else:
             return []
 

--- a/test-data/unit/check-parameter-specification.test
+++ b/test-data/unit/check-parameter-specification.test
@@ -1066,15 +1066,20 @@ def run_job(job: Job[...]) -> T: ...
 [builtins fixtures/tuple.pyi]
 
 [case testTupleAndDictOperationsOnParamSpecArgsAndKwargs]
-from typing import Callable
+from typing import Callable, Iterator, Iterable, TypeVar, Tuple
 from typing_extensions import ParamSpec
 
 P = ParamSpec('P')
+T = TypeVar('T')
+def enumerate(x: Iterable[T]) -> Iterator[Tuple[int, T]]: ...
 
 def func(callback: Callable[P, str]) -> Callable[P, str]:
     def inner(*args: P.args, **kwargs: P.kwargs) -> str:
         reveal_type(args[5])  # N: Revealed type is "builtins.object"
         for a in args:
+            reveal_type(a)  # N: Revealed type is "builtins.object"
+        for idx, a in enumerate(args):
+            reveal_type(idx)  # N: Revealed type is "builtins.int"
             reveal_type(a)  # N: Revealed type is "builtins.object"
         b = 'foo' in args
         reveal_type(b)  # N: Revealed type is "builtins.bool"

